tests/test_flow.py

```diff
@@ -5636,3 +5636,191 @@ def test_flow_reuse_with_cycle():
 
         controller.terminate()
         controller.await_termination()
+
+
+def test_flow_reuse_resets_closeables():
+    """Test that _closeables is properly reset when a flow is reused.
+
+    Baseline test for flow reuse without closeables. This test uses steps (Map, Reduce)
+    that don't create closeables, so it verifies the baseline behavior. Note: This test
+    does NOT detect the ML-11518 bug since there are no closeables to accumulate.
+    """
+    source = SyncEmitSource()
+    map_step = Map(lambda x: x + 1)
+    reduce_step = Reduce(0, lambda acc, x: acc + x)
+
+    source.to(map_step).to(reduce_step)
+
+    first_run_closeables_count = None
+    for run_num in range(3):
+        controller = source.run()
+
+        # Track closeables count - should be consistent across runs
+        if first_run_closeables_count is None:
+            first_run_closeables_count = len(source._closeables)
+        else:
+            assert len(source._closeables) == first_run_closeables_count, (
+                f"Run {run_num}: source._closeables count should be {first_run_closeables_count}, "
+                f"got {len(source._closeables)} (closeables are accumulating!)"
+            )
```

Comment:
This is a very long-winded way of just doing
```suggestion
    for run_num in range(3):
        controller = source.run()
        assert len(source._closeables) == 1
```